### DISCUSSION TITLE
Pull request show commits that are already in the target branch

### DISCUSSION THREAD

**Discussion by jvn-mirkov** [community/community#48132] [Pull Requests]
Hello everyone,

Currently I have this situation:
I had multiple feature branches, dev branch (locked), master branch (locked).
Timeline goes something like this:
t1: dev and master are synced
t2: on multiple feature branches - commits are squashed to one single commit, rebased against the dev, and pushed to dev
t3: after creating the PR dev->master with all changes from [t2], in the commit list there are some commits that were previously merged to master (so before [t1])

I want to merge changes from dev to master but without duplicating commits. I tried the solution that everybody is proposing: change the target in PR to other branch and back to master, but it does not work for me.
Dev is locked so I can't rebase interactivly - remove commits that would be duplicated upon pushing everything to master.

Any ideas how can I fix this?
Thanks!

**Comment by spenserblack** [comment]
Do you have a link to the repo in question, or could you make a minimal reproduction?

How is dev merged into master? Is it a rebase or squash merge?
